### PR TITLE
Docs/restructure tree

### PR DESCRIPTION
**Pull request by aryszka**
Moving all the documentation topics, except for Kubernetes Ingress Controller, Usage and Deployment under a single Reference menu item in order to avoid the breaking the page layout.

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
+    - Reference:
+        - Filters: reference/filters.md
+        - Predicates: reference/predicates.md
+        - Scripts: reference/scripts.md
+        - Plugins: reference/plugins.md
+        - Architecture: reference/architecture.md
+        - Development: reference/development.md
+        - Video - How to build: tutorials/video-howto-build.md
+        - Data Clients:
+            - Eskip File: data-clients/eskip-file.md
```
Comment (arjunrn) [inline mkdocs.yml]: Since the directory was renamed the links to it are broken now. In the `deployment.md` file.

Comment (aryszka) [reply mkdocs.yml]: thanks @arjunrn @szuecs . fixed. Can you please have another look?

Comment (arjunrn) [reply mkdocs.yml]: Looks good now

**Comment by arjunrn** [comment]
👍

**Comment by szuecs** [comment]
:+1:

**Comment by aryszka** [comment]
:+1:

**Pull request merged**